### PR TITLE
Add private groups

### PR DESCRIPTION
**Pull request by jakubgs**
This adds support for Private Group invite URLs.

The code checks for:
- Presence of all three URL arguments: `a`, `a1`, `a2`
- Verifies that `a`(admin key...) is 132 characters long
- Verifies that `a2`(chat key...) is 169 characters long
- Verifies that `a1` does not contain HTML before rendering

Each case is tested, including the valid one.

Resolves: #49

### CODE REVIEW

**Comment by jakubgs** [comment]
One of these days I need to split the tests into separate files.

**Comment by jakubgs** [comment]
Example URL I used to tests:
http://localhost:3000/g/args?a1=Secret%20Club&a=0x0461f576da67dc0bca9888cdb4cb28c80285b756b324109da94a081585ed6f007cf00afede6b3ee5638593674fee100b590318fc7bdb0054b8dd9445acea216ad2&a2=75d9c09b-0d6f-4494-b7da-7bb1e35b1728-0x04f1ab965a3db6433923ceba0488ff605f230aca4c4c93bb6563c10b700cfdbbed2a364c980b9cf9679f352584d4da44cc01fc1e7257f1d2b80ffff5af64aa129f

File: routes/index.js [hunk 1]
```diff
@@ -116,13 +122,47 @@ const handlePublicChannel = (req, res) => {
...
+  try {
+    validatePrivateGroupArgs(req.query)
+  } catch(ex) {
+    handleError(`Invalid private group URL: ${ex.message}`)(req, res)
+    return
+  }
+  let groupName = req.query.a1
+  genPage(req, res, {
+    title: `Join "${groupName}" private group in Status`,
+    info: `Join private group <span class="inline-block align-bottom w-32 truncate">${groupName}</span> in Status.`,
```
Comment (flexsurfer) [inline routes/index.js:158]: in status we call it `Group chat` so probably it should be `Join group chat` ?

Comment (jakubgs) [reply routes/index.js:158]: Oh, I see. Sure, let me change that.

Comment (jakubgs) [reply routes/index.js]: Fixed.

**Comment by jakubgs** [comment]
I've deployed the current state of this PR to prod:
https://join.status.im/g/args?a=0x0461f576da67dc0bca9888cdb4cb28c80285b756b324109da94a081585ed6f007cf00afede6b3ee5638593674fee100b590318fc7bdb0054b8dd9445acea216ad2&a1=Uni%20Links%20Test&a2=d6cea948-34e3-4501-b397-09a4a68b61b7-0x0461f576da67dc0bca9888cdb4cb28c80285b756b324109da94a081585ed6f007cf00afede6b3ee5638593674fee100b590318fc7bdb0054b8dd9445acea216ad2
Maybe @churik and @Serhy can check it out.

**Review by flexsurfer — APPROVED** [review]

**Review by hesterbruikman — APPROVED** [review]
Tried on Android and iOS, with native camera and universal QR scanner. All opened group chat invite in Status 💪💙

**Pull request merged**